From 82f167815470c29287ed2c7efb2afecb22e791fc Mon Sep 17 00:00:00 2001
From: Florian Klink <flokli@flokli.de>
Date: Sat, 28 Mar 2020 18:57:41 -0700
Subject: [PATCH 1308/1852] v4l: xilinx: multi-scaler: fix a printk format
 specifier

commit 57c79e87062aa602053b4b16222942bc7c11b273 from
https://github.com/Xilinx/linux-xlnx.git

Use %pa as printk format specifier for res->start, which is a
resource_size_t, which is basically a phys_addr_t, and pass it by
reference(documented in Documentation/core-api/printk-formats.rst).

```
drivers/media/platform/xilinx/xilinx-multi-scaler.c:2174:15: warning: format '%llx' expects argument of type 'long long unsigned int', but argument 4 has type 'resource_size_t' {aka 'unsigned int'} [-Wformat=]
2174 | dev_dbg(dev, "IO Mem 0x%llx mapped at %p\n", res->start, xm2msc->regs);
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Signed-off-by: Florian Klink <flokli@flokli.de>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index a111ae5bb8a4..3daf90d3377d 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -2171,7 +2171,7 @@ static int xm2msc_parse_of(struct platform_device *pdev,
 	if (IS_ERR((__force void *)xm2msc->regs))
 		return PTR_ERR((__force const void *)xm2msc->regs);
 
-	dev_dbg(dev, "IO Mem 0x%llx mapped at %p\n", res->start, xm2msc->regs);
+	dev_dbg(dev, "IO Mem %pa mapped at %p\n", &res->start, xm2msc->regs);
 
 	ret = of_property_read_u32(node, "xlnx,max-chan",
 				   &xm2msc->max_chan);
-- 
2.31.1

